### PR TITLE
feat: delete EKS node group resources

### PR DESCRIPTION
**Pull request by Paladin-Dranser**
Sometimes it is required to delete only EKS node groups, not EKS cluster
so that **eks-nodegroup** resource type is added to have possibility to work with EKS node group as well.

Add **delete** action for EKS node group resource type.

Issue: #6735

### CODE REVIEW

File: c7n/resources/eks.py
Comment (kapilt) [inline c7n/resources/eks.py:69]: generally our actions don't wait for async operations, was there any particular reason for it here? this might be more appropriate for the test validation code

File: tests/terraform/eks_nodegroup_delete/main.tf
Comment (kapilt) [inline tests/terraform/eks_nodegroup_delete/main.tf:1]: nice! kudos for using the terraform testing support.

**Review by kapilt — COMMENTED** [review]
thanks! looks pretty good to me.

File: tests/terraform/eks_nodegroup_delete/main.tf
Comment (kapilt) [inline tests/terraform/eks_nodegroup_delete/main.tf:2]: we don't want region hard coded here, we want this to flow from environment variables so the test runner can target different regions, terraform uses AWS_REGION

File: c7n/resources/eks.py
Comment (Paladin-Dranser) [reply c7n/resources/eks.py]: I had reused logic of EKS deletion (EKS node groups are deleted because EKS Cluster can't be deleted if node groups exist).

Thanks for info.
Updated.

**Comment by Paladin-Dranser** [comment]
@kapilt,
I see the next message:
_1 workflow awaiting approval
First-time contributors need a maintainer to approve running workflows. Learn more. 4 expected and 5 successful checks_
Could you please to explain how I can get the approval for running other workflows?

P.S. I have had no experience with contribution (especially on GitHub)

**Comment by Paladin-Dranser** [comment]
Converted to **draft** because I found the issue if multiple filters (e.g., **and**) are used.

I will fix issues and update tests to use additional filters.

**Comment by Paladin-Dranser** [comment]
Fixed the issue (there was incorrect id/arn arguments).

Updated test to create 2 EKS node groups and c7n policy deletes only 1 node group (deleted_example) using complex filter (cluster name and tags with **and**).

Cleaned git commit history (remove useless refactor/fix commits).

**Comment by Paladin-Dranser** [comment]
Rebase PR because it was out-of-date with master.

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**
